modMcf/src/org/aion/mcf/config/CfgApiRpc.java

```diff
@@ -115,25 +123,63 @@ public void fromXML(final XMLStreamReader sr) throws XMLStreamException {
                                 e.printStackTrace();
                             }
                             break;
-                        case "threads":
+                        case "worker-threads": {
```

Comment:
this may make old config files incompatible

Reply:
yes, that was the point. Initially, I had both `threads` and `worker-threads` map to the same case, but then I didn't want the user the explicitly set the worker-thread count unless they really knew what they were doing, so I ended up removing the `threads` options altogether. The user would have to read through the RPC config wiki to find the appropriate option to constrain thread count.

Reply:
@ali-sharif do you tested the config file capabilities? Meaning the old version will not block the kernel running.

Reply:
Yes, tested that if `threads` is declared in the config file, the kernel just ignores it and uses the default value, which was the behaviour I wanted to enforce. In general, if an xml node is found in the config but not in the mapper, it's just ignored.

Reply:
Is there a place where you document these breaking changes for things like config files? If I wouldn't have looked into the pull request, I don't know where I could've find this information. Also, given the fact that pools are very dependent on this configuration, we really need to have this information into the open.

Reply:
just to confirm, no breaking changes to the config. we spent a lot of time writing very detailed documentation on the RPC server https://github.com/aionnetwork/aion/wiki/JSON-RPC-API-Docs, which should have been communicated to the pools (we will try to remedy that soon)